packages/sdk/src/base/transport/axios-transport.ts

```diff
@@ -103,16 +96,15 @@ export class AxiosTransport implements ITransport {
 			};
 
 			const token = this._storage.auth_token;
-			const expiration = this._storage.auth_expires;
-			if (options.sendAuthorizationHeaders) {
-				if (token && ((expiration !== null && expiration > Date.now()) || expiration === null)) {
-					if (token.startsWith(`Bearer `)) {
-						config.headers.Authorization = token;
-					} else {
-						config.headers.Authorization = `Bearer ${token}`;
-					}
+			const expiration = this._storage.auth_expires_at;
+			if (token && ((expiration !== null && expiration > Date.now()) || expiration === null)) {
+				// Expires but hasn't yet, or doesn't expire
+				if (token.startsWith(`Bearer `)) {
+					config.headers.Authorization = token;
+				} else {
+					config.headers.Authorization = `Bearer ${token}`;
 				}
-			}
+			} // There is an edge case where it has expired by the time we get here, and refresh either hasn't had enough lead time or is stalled for some reason...
```

Comment:
Like this shouldn't be handling this, but. Idk, there are a lot of specific nitty race condition things that I'm trying to look out for.